klog.go

```diff
@@ -1359,22 +1359,28 @@ func (v Verbose) Infof(format string, args ...interface{}) {
 // See the documentation of V for usage.
 func (v Verbose) InfoS(msg string, keysAndValues ...interface{}) {
 	if v.enabled {
-		logging.infoS(v.logr, v.filter, msg, keysAndValues...)
+		logging.infoS(v.logr, v.filter, 0, msg, keysAndValues...)
 	}
 }
 
+// InfoSDepth acts as InfoS but uses depth to determine which call frame to log.
+// InfoSDepth(0, "msg") is the same as InfoS("msg").
+func InfoSDepth(depth int, msg string, keysAndValues ...interface{}) {
```

Comment:
InfoS interface doesn't cover depth methods as it was based on https://github.com/go-logr/logr/blob/master/logr.go. I don't think we should introduce `InfoSDepth` without having design on `logr` side.

Reply:
`logr` doesn't have any concept of call stacks. I also don't think that it make sense to add something to an interface that is supposed to be conceptually simple.

In `klogr`, the situation is different. It's a specific implementation, not just an interface. It has support for call stacks. To me it makes sense to support them consistently, regardless which of the different API functions is used.

Reply:
The goal was to eventually replace klog.InfoS call-sites with logr.  Anyone who would use logr and wrap it in a helper function would have this same problem. 

We should consider adding it to logr.  I don't think I see a simpler equivalent off the top of my head.  Not every implementation will have that functionality, but they could simply implement is as a pass-thru to Info/Error.

Reply:
> The goal was to eventually replace klog.InfoS call-sites with logr.

That is what can be done with this PR, because then the klogr instance that provides the logr interface will produce correct line numbers and use the same output format as klog.

> Anyone who would use logr and wrap it in a helper function would have this same problem.

That is an unsolved issue: https://github.com/go-logr/logr/issues/30

However, regardless of which direction logr takes, this PR will still be useful to implement that additional API. klog itself doesn't have per-logger depth settings, so some function variants with per-call depth parameters will be needed.

Reply:
To verify the claim that this PR can serve as basis for the future logr extension, I prepared an implementation of the tentative `WithCallDepth` in https://github.com/pohly/klog/commits/klogr-call-depth

Reply:
@serathius logr now has `WithCallDepth` in its master branch, therefore I turned my WIP commit into a PR in #207.